deepinv/physics/mri.py

```diff
@@ -22,86 +24,206 @@ class MRI(DecomposablePhysics):
     The complex images :math:`x` and measurements :math:`y` should be of size (B, 2, H, W) where the first channel corresponds to the real part
     and the second channel corresponds to the imaginary part.
 
-    :param torch.Tensor mask: binary mask.
-        The mask size should be of the form (H,W) where H is the image height and W is the image width.
+    A fixed mask can be set at initialisation, or a new mask can be set either at forward (using ``physics(x, mask=mask)``) or using ``update_parameters``.
+
+    :param torch.Tensor mask: binary mask, where 1s represent sampling locations, and 0s otherwise.
+        The mask size can either be (H,W), (C,H,W), or (B,C,H,W) where H, W are the image height and width, C is channels (typically 2) and B is batch size.
     :param torch.device device: cpu or gpu.
 
     |sep|
 
     :Examples:
 
-        Single-coil MRI operator with 4x acceleration:
+        Single-coil accelerated MRI operator with subsampling mask:
 
         >>> from deepinv.physics import MRI
         >>> seed = torch.manual_seed(0) # Random seed for reproducibility
-        >>> x = torch.randn(1, 2, 3, 3) # Define random 3x3 image
-        >>> mask = torch.ones((3, 3))
-        >>> mask[:, ::2] = 0
-        >>> physics = MRI(mask=mask)
+        >>> x = torch.randn(1, 2, 2, 2) # Define random 2x2 image
+        >>> mask = 1 - torch.eye(2) # Define subsampling mask
+        >>> physics = MRI(mask=mask) # Define mask at initialisation
+        >>> physics(x)
+        tensor([[[[ 0.0000, -1.4290],
+                  [ 0.4564, -0.0000]],
+        <BLANKLINE>
+                 [[ 0.0000,  1.8622],
+                  [ 0.0603, -0.0000]]]])
+        >>> physics = MRI(img_size=x.shape) # No subsampling
         >>> physics(x)
-        tensor([[-0.5305,  0.0351,  0.3326,  2.1730,  1.7072,  0.0418]])
+        tensor([[[[ 2.2908, -1.4290],
+                  [ 0.4564, -0.1814]],
+        <BLANKLINE>
```

Comment:
why the BLANKLINE?

Reply:
Duno, that's the result we get when testing with doctest :

python -m doctest -v deepinv/physics/mri.py